src/hooks/useSAMLSignIn.ts

```diff
@@ -12,9 +39,28 @@ export const useSAMLSignIn = (samlAuthProviderId?: string) => {
       samlAuthProviderId
     );
 
+    if (showLoginLoading) {
+      showLoginLoading();
+    }
+
     firebase
       .auth()
       .signInWithPopup(SAMLAuthProvider)
+      .then((result) => {
+        if (
+          !result.additionalUserInfo?.isNewUser ||
+          !result.user?.uid ||
+          !samlProfileMappings
+        )
+          return;
+
+        const mappedProfileData = getMappedValues(samlProfileMappings, result);
+
+        setLocalStorageItem(
+          LocalStorageItem.prefillProfileData,
+          mappedProfileData
+        );
```

Comment:
I believe we can read the provided data as claims off the firebase auth identity token always can't we? So do we even need to use `localStorage` like this?

Reply:
See further discussion in https://github.com/sparkletown/sparkle/pull/1667#discussion_r659470758

Reply:
Done in https://github.com/sparkletown/sparkle/pull/1667/commits/c77cd255fb597da0dd2611bc66089e77dbe0e59f